crates/bin/node/src/grpc_service.rs

```diff
@@ -554,6 +555,25 @@ impl Node for GrpcState {
         Ok(Response::new(AcknowledgeResponse {}))
     }
 
+    async fn check_state(
+        &self,
+        request: Request<CheckStateRequest>,
+    ) -> Result<Response<CheckStateResponse>, Status> {
+        let ys: Vec<Vec<u8>> = request.into_inner().ys;
```

Comment:
Do a lilttle bit of input validation here, by making sure that all the Ys value are valid `PublicKey`, and pass a Vec<PublicKey> to the `inner` method.

It's a bit of extra processing but it's more future proof and expressive